packages/common/src/types/index.ts

```diff
@@ -1,14 +1,21 @@
+import { ObjectWithStringKeys } from './utility';
+
 export * from './utility';
 
-export type Item = {
+type RecordWithId = { id: string };
+
+export interface DomainObject extends RecordWithId, ObjectWithStringKeys {}
```

Comment:
This is quite silly but having the assurance that there was a key `id` helped in a few places.

The name needs help and maybe just `RecordWithId` would be better